modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -405,6 +406,12 @@ const FrontendDataSet = ({
 			});
 	}
 
+	function updatePagination(dataSetData) {
+		if (!dataSetData.items.length && dataSetData.totalCount > 0) {
+			setPageNumber(1);
```

Comment:
Resetting page number to 1 is the simplest solution, and does fix the issue. But, do we maybe want to select the last page instead of first? Since we deleted the last item, it may be more expected. A product question, pinging @herrardo @dsanz

Reply:
Yep, we could easily calculate the previous page and set it instead of defaulting always to the first one.

Reply:
I've realized that we also have available the `.lastPage` within the request, so there is no need to add extra logic to calculate the new page.

Need to inline pagination update logic because, even though **ci** does not complain about `react-hooks/exhaustive-deps`, locally I have a couple of errors due to `React Hook useEffect has a missing dependency: 'updateDataSetItems'.`

Reply:
Well, I'd say taking user to the last page looks better in terms of navigation "disruptiveness". We must consider that, from the last moment FDS fetched that page, many deletions might have taken place, so last page might not be just the previous one.

What I mean is that `setPageNumber(pageNumber-1)` might not work. Once we get a response with an empty array, we can do something like `setPageNumber(Math.trunc(totalCount/paginationDelta)`, so that it directly fetches the last page.

Even in that scenario, in presence of a massive deletion we could get yet another empty page, but this solution should converge faster to the last valid page as compared to substract 1 on each new request.